be/src/exprs/string_functions.cpp

```diff
@@ -1775,16 +1794,41 @@ static const char* skip_trailing_spaces(const char* begin, const char* end) {
         p += SSE2_BYTES;
     }
 #endif
-    for (--p; p >= begin && *p == ' '; --p) {
+    if constexpr (trim_single) {
+        if (remove[0] == ' ') {
```

Comment:
why need special check for ' '? just using remove[0] is ok?

Reply:
For correctness it's enough. For performance it may save some memory loadings